packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs

```diff
@@ -851,6 +852,82 @@ where
     }
 }
 
+impl PostgresStore<Transaction<'_>> {
+    /// Inserts the specified [`EntityId`]s into the database.
+    ///
+    /// # Errors
+    ///
+    /// - if inserting the [`EntityId`]s failed.
+    async fn insert_entity_ids(
+        &self,
+        entity_ids: impl IntoIterator<Item = EntityId, IntoIter: Send> + Send,
+    ) -> Result<u64, InsertionError> {
+        let sink = self
+            .client
+            .copy_in("COPY entity_ids (entity_id) FROM STDIN BINARY")
+            .await
+            .into_report()
+            .change_context(InsertionError)?;
+        let writer = BinaryCopyInWriter::new(sink, &[Type::UUID]);
```

Comment:
Does writing end when the writer (or sink) is dropped? I wonder if this would also introduce a point of serial execution across a single postgres client as we saw with the structural queries. In that case, we'd have to make sure not to execute/poll multiple bulk operations at once because of the IO writing here

Reply:
The writing ends when calling `writer.finish()` https://github.com/hashintel/hash/blob/a5f2b4cc39e48f2c107d48ac717bc47abb2119f4/packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs#L884-L888

[Link to the documentation](https://docs.rs/tokio-postgres/latest/tokio_postgres/binary_copy/struct.BinaryCopyInWriter.html) for reference